ui/text_displayer.js

```diff
@@ -169,130 +169,128 @@ shaka.ui.TextDisplayer = class {
+   * @private
+   */
+  buildCueElements_(tokens) {
+    const supportedTags = [
+      'i',
+      'b',
+      'u',
+    ];
+
+    const elements = [];
+
+    // Build a set of root elements from the token collection.
+    const stack = [];
+    for (const currentToken of tokens) {
+      const supportedTag = supportedTags.includes(currentToken.value);
+      if (currentToken.type === 'text' || !supportedTag) {
+        // Element is a text field or an unsupported tag.
+        const textElement = document.createElement('span');
+        textElement.textContent = currentToken.value;
+        if (stack.length > 0) {
+          // Add to bottom stack element.
+          stack[stack.length - 1].appendChild(textElement);
         } else {
-          if (currentToken.type === 'startTag') {
-            // Add this new element to the stack.
-            stack.push(document.createElement(currentToken.value));
-            if (stack.length > 1) {
-              // Add this item as a child of the element one up the stack.
-              stack[stack.length - 2].appendChild(stack[stack.length - 1]);
-            }
-          } else if (currentToken.type === 'endTag') {
-            // Check to see if this closes the last opened tag.
-            const bottomName = stack[stack.length - 1].tagName.toLowerCase();
-            if (bottomName === currentToken.value.toLowerCase()) {
-              // Close this tag off the stack.
-              const v = stack.pop();
-              if (stack.length === 0) {
-                // Stack is empty, push root of this tree into elements.
-                elements.push(v);
-              }
-            } else {
-              // Input syntax error.
+          // Add as standalone.
+          elements.push(textElement);
+        }
+      } else {
+        if (currentToken.type === 'startTag') {
+          // Add this new element to the stack.
+          stack.push(document.createElement(currentToken.value));
+          if (stack.length > 1) {
+            // Add this item as a child of the element one up the stack.
+            stack[stack.length - 2].appendChild(stack[stack.length - 1]);
+          }
+        } else if (currentToken.type === 'endTag') {
+          // Check to see if this closes the last opened tag.
+          const bottomName = stack[stack.length - 1].tagName.toLowerCase();
+          if (bottomName === currentToken.value.toLowerCase()) {
+            // Close this tag off the stack.
+            const v = stack.pop();
+            if (stack.length === 0) {
+              // Stack is empty, push root of this tree into elements.
+              elements.push(v);
             }
+          } else {
+            // Input syntax error.
           }
         }
       }
-      // If we have unclosed tags, add the unclosed elements.
-      if (stack.length > 0) {
-        elements.push(stack[0]);
-      }
-
-      return elements;
-    };
+    }
+    // If we have unclosed tags, add the unclosed elements.
```

Comment:
At first I thought this was a mistake, since it would only add one element, but there might be several in the array.  It is probably worth an extra comment to prevent confusion.  Something like this would have helped me understand it more quickly:

```js
// Since the elements in the stack are already nested inside each other,
// we only need to push stack[0].
```

Reply:
I agree. It seems obvious now with the logic flow in my organic cache, but I wonder if it would still be so clear a year from now.

Your example comment has been added.
https://github.com/google/shaka-player/blob/0292863c595d75c16cce69e99322e6bf79017ddd/ui/text_displayer.js#L288-L293